### PR TITLE
Make better use of AssertJ's string matchers

### PR DESCRIPTION
**Pull request by rhowe**
Where we are using `String#format(...)` to deal with platform-specific line endings, we can instead use `isEqualToNormalizingNewlines`.

We can also replace some calls to `String#format(String...)` with AssertJ's built-in support for format strings.

### CODE REVIEW

**Comment by rhowe** [comment]
For the migration command help text output tests, what do you think about moving them to files in `src/test/resources/`, at least until Java 17...

File: dropwizard-migrations/src/test/java/io/dropwizard/migrations/DbMigrateCommandTest.java [hunk 1]
```diff
@@ -61,10 +61,10 @@ void testDryRun() throws Exception {
...
         migrateCommand.setOutputStream(new PrintStream(baos));
         migrateCommand.run(null, new Namespace(Collections.singletonMap("dry-run", true)), conf);
-        assertThat(baos.toString(UTF_8.name())).startsWith(String.format(
-                "-- *********************************************************************%n" +
-                "-- Update Database Script%n" +
-                "-- *********************************************************************%n"));
+        assertThat(baos.toString(UTF_8.name())).isEqualToNormalizingNewlines(
+                "-- *********************************************************************\n" +
+                "-- Update Database Script\n" +
+                "-- *********************************************************************\n");
```
Comment (joschi) [inline dropwizard-migrations/src/test/java/io/dropwizard/migrations/DbMigrateCommandTest.java]: Incorrect replacement (starts with vs. equals).

Comment (rhowe) [reply dropwizard-migrations/src/test/java/io/dropwizard/migrations/DbMigrateCommandTest.java]: Right you are - fun that the test still passed!

File: dropwizard-migrations/src/test/java/io/dropwizard/migrations/DbClearChecksumsCommandTest.java [hunk 2]
```diff
@@ -31,23 +31,23 @@ void testRun() throws Exception {
...
-                "named arguments:%n" +
-                "  -h, --help             show this help message and exit%n" +
-                "  --migrations MIGRATIONS-FILE%n" +
-                "                         the file containing  the  Liquibase migrations for%n" +
-                "                         the application%n" +
-                "  --catalog CATALOG      Specify  the   database   catalog   (use  database%n" +
-                "                         default if omitted)%n" +
-                "  --schema SCHEMA        Specify the database schema  (use database default%n" +
-                "                         if omitted)%n"));
+        assertThat(out.toString(UTF_8.name())).isEqualTo(
```
Comment (zUniQueX) [inline dropwizard-migrations/src/test/java/io/dropwizard/migrations/DbClearChecksumsCommandTest.java]: Looks like there is a NormalizingNewlines missing

Comment (rhowe) [reply dropwizard-migrations/src/test/java/io/dropwizard/migrations/DbClearChecksumsCommandTest.java]: Right you are. I think I was flipping between `hasToString()` and this and ended up broken in the middle somewhere!

**Review by joschi — APPROVED** [review]

**Pull request merged**